spec/lang/types.md

```diff
@@ -187,5 +187,14 @@ impl IntType {
     pub fn bring_in_bounds(&self, i: Int) -> Int {
         i.bring_in_bounds(self.signed, self.size)
     }
+
+    /// Generate the return type for IntWithOverflow
+    pub fn overflow_type<T: Target>(&self) -> Type {
+        let self_ty = Type::Int(*self); 
+        let fields = list![(Size::ZERO, self_ty), (self_ty.size::<T>(), Type::Bool)];
+        let size = self_ty.size::<T>() + self_ty.size::<T>();
```

Comment:
This does not seem right -- if `self_ty` is `u16`, size must be 4 or else it is not a multiple of the alignment. Currently you are generating a not-well-formed type in that case.

Reply:
Maybe I'm missing something but `self_ty.size::<T>()  = 2` and `self_ty.size::<T>() + self_ty.size::<T>() = 4`.
And double the size is always a multiple of the alignment.

Reply:
tried to increase the clarity by changing the line to:
```rust
let size = self_ty.size::<T>() * Int::from(2); 
```

Reply:
d'oh, I misread.
Please add a comment explaining all that is going on here -- all this only makes sense because the 2nd field has alignment 1 and size 1.